src/restLink.ts

```diff
@@ -143,7 +214,40 @@ export const validateRequestMethodForOperationType = (
 
 let exportVariables = {};
 
-const resolver: Resolver = async (fieldName, root, args, context, info) => {
+/** Apollo-Link getContext, provided from the user & mutated by upstream links */
+interface LinkChainContext {
+  /** Credentials Policy for Fetch */
+  credentials?: RequestCredentials | null;
+
+  /** Headers the user wants to set on this request. See also headersMergePolicy */
+  headers?: RestLink.InitializationHeaders | null;
+
+  /** Will default to concatHeadersMergePolicy unless headersToOverride is set */
+  headersMergePolicy?: RestLink.HeadersMergePolicy | null;
+
+  /** List of headers to override, passing this will swap headersMergePolicy if necessary */
+  headersToOverride?: string[] | null;
+}
+
+/** Context passed via graphql() to our resolver */
+interface RequestContext {
+  /** Headers the user wants to set on this request. See also headersMergePolicy */
+  headers: Headers;
+
+  /** Credentials Policy for Fetch */
+  credentials?: RequestCredentials | null;
+
+  endpoints: RestLink.Endpoints;
+  customFetch: RestLink.CustomFetch;
+}
+
+const resolver: Resolver = async (
+  fieldName: string,
+  root: any,
+  args: any,
+  context: RequestContext,
+  info: any,
```

Comment:
I believe the `Resolver` type provides these types. Also, I believe `info should be `ExecInfo`? 

https://github.com/apollographql/apollo-client/blob/26bc85923962ffa5fac53a674df58acacd66949f/packages/graphql-anywhere/src/graphql.ts#L23-L29

Reply:
Thanks @Skovy! I wasn't sure where to go for other examples of the correct types here. I needed the pointer!

Reply:
👍 no problem! Do you think we should remove the `Resolver` import I added since we are now defining the entire type signature, but with a bit more specificity now that we have `RequestContext`? It might be a bit redundant?

Reply:
I'm comfortable with leaving Resolver in, that provides more links for a person reading the code to understand what's going on and who owns the shape of that method. -- RequestContext is a helper for us internally, while Resolver specifies that as any, so both are useful in my opinion.

Happy to see it change if someone feels strongly about it!

Reply:
Oh, that is a great point! I did have to dig a bit to find the original `Resolver` types 😁 :shipit: